### PR TITLE
tweak: randomly choose special kit receiver

### PR DESCRIPTION
**Pull request by Gaxeer**
## What Does This PR Do
Steal object special kits are now given to random objective owner (multiple owners possible when objective is assigned to team). 
Previously the SAME item was given to each one in team, which effectively resulted in item being given to the last owner.

## Why It's Good For The Game
More predictable special equipment giving strategy.

## Images of changes
All in code.

## Testing
Spawned, given objective.

## Changelog
:cl:
tweak: Steal object special kits are now given to random objective owner. Previously the SAME item was given to each one in team, which effectively resulted in item being given to the last owner.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -567,16 +567,22 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 /datum/objective/steal/proc/select_target()
 	var/list/possible_items_all = GLOB.potential_theft_objectives+"custom"
 	var/new_target = input("Select target:", "Objective target", null) as null|anything in possible_items_all
-	if(!new_target) return
+	if(!new_target)
+		return
+
 	if(new_target == "custom")
 		var/datum/theft_objective/O=new
```
Comment (Contrabang) [inline code/game/gamemodes/objective.dm:574]: While you're here, this O needs to be qdeleted for the early returns on 576 and 583

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -599,23 +605,33 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
-			to_chat(H, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
-		else
-			to_chat(H, "<span class='userdanger'>Unfortunately, you weren't able to get a stealing kit. This is very bad and you should adminhelp immediately (press F1).</span>")
-			message_admins("[ADMIN_LOOKUPFLW(H)] Failed to spawn with their [item_path] theft kit.")
-			qdel(I)
+			to_chat(kit_receiver, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
+			return
+
+		to_chat(kit_receiver, "<span class='userdanger'>Unfortunately, you weren't able to get a stealing kit. This is very bad and you should adminhelp immediately (press F1).</span>")
+		message_admins("[ADMIN_LOOKUPFLW(kit_receiver)] Failed to spawn with their [item_path] theft kit.")
```
Comment (Contrabang) [inline code/game/gamemodes/objective.dm:632]: IMO, this should first loop through all members before giving up and messaging admins.

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -599,23 +605,33 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
+			continue
+
+		var/where = kit_receiver.equip_in_one_of_slots(item_to_give, slots)
 		if(where)
-			to_chat(H, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
-		else
-			to_chat(H, "<span class='userdanger'>Unfortunately, you weren't able to get a stealing kit. This is very bad and you should adminhelp immediately (press F1).</span>")
-			message_admins("[ADMIN_LOOKUPFLW(H)] Failed to spawn with their [item_path] theft kit.")
-			qdel(I)
+			to_chat(kit_receiver, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
```
Comment (Contrabang) [inline code/game/gamemodes/objective.dm:628]: IMO, if there are multiple objective owners, we might want to tell other members of the objective that someone else on their team has the objective kit.

File: code/game/gamemodes/objective.dm (hunk 2 above)
Comment (Gaxeer) [reply code/game/gamemodes/objective.dm:632]: But players should be notified each iteration, what do you think?

Comment (Contrabang) [reply code/game/gamemodes/objective.dm:632]: If it cant assign it to the first one, but can assign it to the second one, then there is no need for them to contact admins, it only really matters if none of them can get it

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/gamemodes/objective.dm [hunk 4]
```diff
@@ -599,23 +604,46 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
+		var/where = kit_receiver.equip_in_one_of_slots(item_to_give, slots)
+		if(!where)
+			continue
+
+		to_chat(kit_receiver, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
+		for(var/datum/mind/objective_owner as anything in objective_owners)
+			if(kit_receiver_mind == objective_owner || !objective_owner.current)
+				continue
+
+			to_chat(objective_owner.current, "<br><br>[kit_receiver] has received box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
```
Comment (Contrabang) [inline code/game/gamemodes/objective.dm:634]:
```suggestion
			to_chat(objective_owner.current, "<br><br>[kit_receiver] has received a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
one tiny thing

**Review by Contrabang — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**